### PR TITLE
Update dependencies 15-09-2020

### PR DESCRIPTION
**Pull request by veado**
## tl;dr 

This update failed due several dependency issues and will be postponed until `react-scripts` supports `typescript@4`

## Available updates

```
ncu
Checking asgardex-electron/package.json
[====================] 89/89 100%

 @ant-design/icons                           ^4.2.1  →         ^4.2.2   
 @devexperts/remote-data-ts                  ^2.0.3  →         ^2.0.4   
 @openapitools/openapi-generator-cli  ^1.0.15-4.3.1  →  ^1.0.18-4.3.1   
 antd                                        ^4.6.1  →         ^4.6.4   
 electron-log                                ^4.2.2  →         ^4.2.4   
 ethers                                      ^5.0.9  →        ^5.0.13   
 fp-ts                                       ^2.8.1  →         ^2.8.2   
 observable-hooks                            ^3.0.0  →         ^3.1.2   
 react-intl                                  ^5.4.5  →         ^5.8.1   
 react-scripts                                3.4.1  →          3.4.3   
 rxjs                                        ^6.6.2  →         ^6.6.3   
 styled-components                           ^5.1.1  →         ^5.2.0   
 typescript                                  ^3.9.7  →         ^4.0.2   
 @storybook/addon-actions                   ^5.3.19  →        ^6.0.21   
 @storybook/addon-knobs                     ^5.3.19  →        ^6.0.21   
 @storybook/addon-links                     ^5.3.19  →        ^6.0.21   
 @storybook/addon-viewport                  ^5.3.19  →        ^6.0.21   
 @storybook/addons                          ^5.3.19  →        ^6.0.21   
 @storybook/react                           ^5.3.19  →        ^6.0.21   
 @storybook/theming                         ^5.3.19  →        ^6.0.21   
 @testing-library/jest-dom                  ^5.11.2  →        ^5.11.4   
 @testing-library/react                     ^10.4.7  →        ^11.0.4   
 @testing-library/user-event                ^12.1.0  →        ^12.1.5   
 @types/jest                                ^26.0.8  →       ^26.0.13   
 @types/node                                    ^13  →            ^14   
 @types/react                              ^16.9.44  →       ^16.9.49   
 @types/styled-components                    ^5.1.2  →         ^5.1.3   
 @typescript-eslint/eslint-plugin            ^3.7.1  →         ^4.1.1   
 @typescript-eslint/parser                   ^3.7.1  →         ^4.1.1   
 concurrently                                ^5.2.0  →         ^5.3.0   
 electron                                    ^9.1.2  →        ^10.1.2   
 eslint-plugin-react                        ^7.20.5  →        ^7.20.6   
 eslint-plugin-react-hooks                   ^4.0.8  →         ^4.1.2   
 husky                                       ^4.2.5  →         ^4.3.0   
 lint-staged                               ^10.2.11  →        ^10.3.0   
 prettier                                    ^2.0.5  →         ^2.1.1   
 testcafe                                    ^1.8.8  →         ^1.9.2   
 ts-jest                                    ^26.1.4  →        ^26.3.0   
 ts-loader                                   ^8.0.2  →         ^8.0.3   
 ts-node                                    ^8.10.2  →         ^9.0.0   
 wait-on                                     ^5.1.0  →         ^5.2.0  
```

## Issues 

*(1)* `typescript@4.x` is not fully supported by latest `react-scripts@3.4.3` (https://github.com/facebook/create-react-app/issues/9515). It causes `eslint` issues and needs to be updated to latest `@typescript-eslint/eslint-plugin@4.x` before. This needed change is already in master (https://github.com/facebook/create-react-app/pull/9434), but not published yet.

*(2)* `testcafe` does not support `electron@10.x`. we will still stick with `electron@9.x`.

*(3)* Update to latest `fp-ts@2.8.2` failed, it introduces type errors by using `@devexperts/remote-data-ts@2.0.4` 

```
asgardex-electron/src/renderer/helpers/fpHelpers.ts
[dev:app] TypeScript error in asgardex-electron/src/renderer/helpers/fpHelpers.ts(18,38):
[dev:app] No overload matches this call.
[dev:app]   The last overload gave the following error.
[dev:app]     Argument of type 'Monad2<"RemoteData"> & Foldable2<"RemoteData"> & Traversable2<"RemoteData"> & Bifunctor2<"RemoteData"> & Alt2<"RemoteData"> & Extend2<...> & Alternative2<...>' is not assignable to parameter of type 'Apply<"RemoteData">'.
[dev:app]       Types of property 'ap' are incompatible.
[dev:app]         Type '<E, A, B>(fab: RemoteData<E, (a: A) => B>, fa: RemoteData<E, A>) => RemoteData<E, B>' is not assignable to type '<A, B>(fab: HKT<"RemoteData", (a: A) => B>, fa: HKT<"RemoteData", A>) => HKT<"RemoteData", B>'.
[dev:app]           Types of parameters 'fab' and 'fab' are incompatible.
[dev:app]             Type 'HKT<"RemoteData", (a: A) => B>' is not assignable to type 'RemoteData<unknown, (a: unknown) => unknown>'.
[dev:app]               Type 'HKT<"RemoteData", (a: A) => B>' is missing the following properties from type 'RemoteSuccess<(a: unknown) => unknown>': _tag, value  TS2769
[dev:app] 
[dev:app]     16 |   return sequenceTOption(first, ...rest)
[dev:app]     17 | }
[dev:app]   > 18 | export const sequenceTRD = sequenceT(RD.remoteData)
[dev:app]        |                                      ^
[dev:app]     19 | export const sequenceTRDFromArray = <E, A>(onEmpty: () => E) => ([first, ...rest]: RD.RemoteData<
```

^ These errors are very similar as described in https://github.com/devexperts/remote-data-ts/issues/49 

Due lack of time we will stick on `fp-ts@2.8.1` for now.

### CODE REVIEW

**Comment by veado** [comment]
We will give another try when `react-scripts` supports `typescript@4.x`. _Closing now..._
